Core/GDCore/IDE/InstructionValidator.cpp

```diff
@@ -0,0 +1,141 @@
+    if (!expressionValidator.GetAllErrors().empty()) {
+      return false;
+    }
+    // New object variable instructions require the variable to be
+    // declared while legacy ones don"t.
+    // This is why it"s done here instead of in the parser directly.
+    if (parameterType == "objectvar" &&
+        gd::VariableInstructionSwitcher::IsSwitchableVariableInstruction(
+            instruction.GetType())) {
+      // Check at least the name of the root variable, it"s the best we can
+      // do.
+      auto &objectsContainersList =
+          projectScopedContainers.GetObjectsContainersList();
+      auto &objectName = instruction.GetParameter(0).GetPlainString();
+      auto &variableName =
+          instruction.GetParameter(parameterIndex).GetPlainString();
+      if (objectsContainersList.HasObjectOrGroupWithVariableNamed(
+              objectName,
+              gd::InstructionValidator::GetRootVariableName(variableName)) ==
+          gd::ObjectsContainersList::DoesNotExist) {
+        return false;
+      }
+    }
+  } else if (gd::ParameterMetadata::IsObject(parameterType)) {
+    auto &objectOrGroupName =
+        instruction.GetParameter(parameterIndex).GetPlainString();
+    auto &objectsContainersList =
+        projectScopedContainers.GetObjectsContainersList();
+    return objectsContainersList.HasObjectOrGroupNamed(objectOrGroupName) &&
+           (parameterMetadata.GetExtraInfo().empty() ||
+            objectsContainersList.GetTypeOfObject(objectOrGroupName) ==
+                parameterMetadata.GetExtraInfo()) &&
+           InstructionValidator::HasRequiredBehaviors(
+               instruction, metadata, parameterIndex, objectsContainersList);
+  } else if (gd::ParameterMetadata::IsExpression("resource", parameterType)) {
+    auto &resourceName =
+        instruction.GetParameter(parameterIndex).GetPlainString();
+    return projectScopedContainers.GetResourcesContainersList()
+        .HasResourceNamed(resourceName);
+  }
+  return true;
+}
+
+gd::String InstructionValidator::GetRootVariableName(const gd::String &name) {
+  auto dotPosition = name.find('.');
+  auto squareBracketPosition = name.find('[');
+  if (dotPosition == gd::String::npos &&
+      squareBracketPosition == gd::String::npos) {
+    return name;
+  }
+  return name.substr(0, dotPosition < squareBracketPosition
+                            ? dotPosition
+                            : squareBracketPosition);
+};
+
+bool InstructionValidator::HasRequiredBehaviors(
+    const gd::Instruction &instruction,
+    const gd::InstructionMetadata &instructionMetadata,
+    std::size_t objectParameterIndex,
+    const gd::ObjectsContainersList &objectsContainersList) {
+  auto &objectOrGroupName =
+      instruction.GetParameter(objectParameterIndex).GetPlainString();
+  for (size_t index = objectParameterIndex + 1;
+       index < instructionMetadata.GetParametersCount(); index++) {
+    auto &behaviorParameter = instructionMetadata.GetParameter(index);
```

Comment:
Here and below, try to const everything:

```suggestion
    const auto &behaviorParameter = instructionMetadata.GetParameter(index);
```